File: realtime-server/src/main/java/com/slowhigh/kairos/timer/dto/TimestampAddedEvent.java
package com.slowhigh.kairos.timer.dto;

import java.time.Instant;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "'timestampAdded' 이벤트의 데이터 형식")
public class TimestampAddedEvent {

  @Schema(description = "신규 기준 시간", example = "2025-07-08T15:00:00Z")
  private Instant targetTime;

  @Schema(description = "캡처된 시간", example = "2025-07-08T15:00:00Z")
  private Instant capturedAt;

}

File: server/apps/sync-service/src/main/kotlin/com/sharetimer/syncservice/adapter/output/web/dto/TimestampAddedEvent.kt
package com.sharetimer.syncservice.adapter.output.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.time.Instant

@Schema(description = "Data format for 'timestampAdded' event")
data class TimestampAddedEvent(
    @field:Schema(description = "New target time", example = "2025-07-08T15:00:00Z")
    val targetTime: Instant,
    @field:Schema(description = "Captured time", example = "2025-07-08T15:00:00Z")
    val capturedAt: Instant,
)
